src/assets/js/protocol/protocol-menu.js

```diff
@@ -0,0 +1,233 @@
+        if (animate) {
+            el.classList.add('animate');
+        }
+
+        el.classList.add('selected');
+        el.setAttribute('aria-selected', true);
+        el.setAttribute('aria-expanded', true);
+
+        if (typeof _options.onMenuOpen === 'function') {
+            _options.onMenuOpen(el);
+        }
+    };
+
+    Menu.close = function() {
+        // On small screens more than one menu can be open at the same time.
+        var current = document.querySelectorAll('.mzp-c-menu-category.selected');
+
+        for (var i = 0; i < current.length; i++) {
+            current[i].classList.remove('animate');
+            current[i].classList.remove('selected');
+            current[i].setAttribute('aria-selected', false);
+            current[i].setAttribute('aria-expanded', false);
+        }
+
+        if (typeof _options.onMenuClose === 'function' && current.length > 0) {
+            _options.onMenuClose();
+        }
+
+        return current.length > 0;
+    };
+
+    Menu.onCloseButtonClick = function(e) {
+        e.preventDefault();
+
+        if (typeof _options.onMenuButtonClose === 'function') {
+            _options.onMenuButtonClose();
+        }
+
+        Menu.close();
+    };
+
+    Menu.toggle = function(el) {
+        var state = el.classList.contains('selected') ? true : false;
+
+        if (!state) {
+            Menu.open(el);
+        } else {
+            Menu.close();
+        }
+    };
+
+    Menu.onMouseEnter = function(e) {
+        clearTimeout(_hoverTimeout);
+
+        // Only open the panel if the user shows hover intent
+        _hoverTimeout = setTimeout(function() {
+            var current = Menu.close();
+            var animate = current ? false: true;
+
+            Menu.open(e.target, animate);
+        }, 150);
+    };
+
+    Menu.onMouseLeave = function() {
+        // Clear hover intent timer.
+        clearTimeout(_hoverTimeout);
+
+        _hoverTimeout = setTimeout(function() {
+            Menu.close();
+        }, 150);
+    };
+
+    Menu.onFocusOut = function() {
+        var self = this;
+
+        /**
+         * After an element loses focus, `document.activeElement` will always be `body` before
+         * moving to the next element. A `setTimeout` of `0` circumvents this issue as it
+         * re-queues the JavaScript to run at the end of the current excecution.
+         */
+        setTimeout(function() {
+            if (!self.contains(document.activeElement)) {
+                Menu.close();
+            }
+        }, 0);
+    };
+
+    Menu.onClickWide = function(e) {
+        e.preventDefault();
+        Menu.close();
+        Menu.open(e.target.parentNode);
+    };
+
+    Menu.onClickSmall = function(e) {
+        e.preventDefault();
+        Menu.toggle(e.target.parentNode);
+    };
+
+    Menu.isWideViewport = function() {
+        return _mqWideNav.matches;
+    };
+
+    Menu.handleState = function() {
+        _mqWideNav = matchMedia('(min-width: 768px)');
```

Comment:
Breakpoints should be global variables.

Reply:
I don't think it needs to be global, but I can certainly move it to a variable.